frontend/src/components/common/Label.tsx

```diff
@@ -155,7 +155,11 @@ export interface HoverInfoLabelProps {
   iconProps?: Omit<IconProps, 'icon'>;
 }
 
+// using display inline-block to prevent the label from rendering away from the icon
 const useHoverInfoLabelStyles = makeStyles({
+  display: {
+    display: 'inline-block',
+  },
```

Comment:
You could merge this rule with the noWrap, if the noWrap and this one are always used together.